testsuite/tests/ui/master/test_tenant.py

```diff
@@ -119,3 +119,25 @@ def test_resume_tenant(master_login, navigator, master_threescale, custom_tenant
 
     assert_displayed_in_new_tab(browser, detail_view.open_public_domain, LandingView)
     assert_displayed_in_new_tab(browser, detail_view.open_admin_domain, LoginView)
+
+
+def test_impersonate_tenant(master_login, navigator, master_threescale, custom_tenant, browser):
+    """
+    Test check impersonation functionality:
+        - create tenant
+        - login into master portal
+        - impersonate new created tenant
+        - assert that the impersonation was successful
+            - dashboard is displayed
+            - url is correct
+    """
+    tenant = custom_tenant()
+    tenant.wait_tenant_ready()
```

Comment:
This functionality can be moved to separate fixture as we do for other `custom_entities`. Also `wait_tenant_ready()` is already called in the `custom_tenant()` function.